### ISSUE TITLE
Epetra.Operator

### ISSUE THREAD

**Issue by rveltz** [barche/Trilinos.jl#1]
Hi,

Thank you a lot for your wrapper! I was just wondering if you know a way to interface `Epetra.operator` for which you have to supply a class with function `Apply` ?

Do you think it is possible?

Thank you,

Best regards

Romain

**Comment by barche** [comment]
We could add a C++ class implementing Epetra_Operator that takes a Julia function and calls that when Apply is called, so this would allow you to write Apply in Julia if that is what you would like to do? Assuming the work done in Apply is much greater than the overhead of calling a Julia function from C, this should be viable.

I wasn't really intending to wrap Epetra, but this would be fairly easy to add.

**Comment by rveltz** [comment]
We have to write a wrapper for these Trilinos classes, do we?
In fact, I am interested in NOX-LOCA-Epetra. It seems a lot of work...

**Comment by barche** [comment]
True, that would require wrapping more classes. Do you know if your application would work through Thyra? I already wrapped the classes needed to get Thyra operators and vectors from Tpetra matrices and vectors.

**Comment by rveltz** [comment]
I have checked and it seems it would be difficult. I am interested in NOX-LOCA and I do in PyTrilinos. It seems Thyra is not wrapped for PyTrilinos which means I would need to do the example in c++...

**Comment by barche** [comment]
OK, then in the short term all I could do is help you get started writing the wrappers yourself. I currently don't have an application that would need NOX and didn't intend to wrap Epetra at all (in favor of the more modern Tpetra). My focus now is on getting some linear solvers working in a Julia-friendly way.
